### PR TITLE
(refactor)(performance)(fix): Audit Product Images and update to always use optimized versions

### LINKED ISSUE #3637: Product Images are using full res image and not the processed versions
When I was working on this [PR](https://github.com/reactioncommerce/reaction/pull/3631) I started to notice the full resolution product image was being used in almost every instance. I thought it was only in my branch but I've tested it on master and have seen it every time. Can someone test and confirm I'm not crazy?

### Expected behavior
- Product images should be referencing the appropriate image crop in the `src` attribute.
- For example, product images on the invoice page should be using the `thumbnail` crop
- The image `src` path should look something like this `/assets/files/Media/Z5v7ARFEjWKdoSymL/shoe.jpg?token=eyJhdXRoVG9rZW4iOiJwVnVKS2J5WVRjcG1yOG03QU4ybk15ZkFPb2VXUzJjaGh6Y0VERHZwckxvIn0%3D&store=thumbnail`

### Actual behavior
- Product images are only referencing the full resolution uploaded image
- Product image `src` path looks like this `/assets/files/Media/Z5v7ARFEjWKdoSymL/shoe.jpg?token=eyJhdXRoVG9rZW4iOiJwVnVKS2J5WVRjcG1yOG03QU4ybk15ZkFPb2VXUzJjaGh6Y0VERHZwckxvIn0%3D` without the `&store={size}`

### Steps to reproduce the behavior
- Run `reaction reset` to get a fresh reaction store.
- Login as admin and upload new product images to the sample product
- Upload images to each product variant
- Check the `src` and image info of the new product images to see it's referencing the original uploaded image and not the cropped version
- In another browser purchase both product variants as a guest
- On the invoice page, inspect the product thumbnails to see it referencing the full image and not the thumbnail version

**screenshots**
product-detail
example of small image referencing the og image

<br/>

product-varient
product variant images referencing the og image and not the small square version

<br/>
<br/>

invoice-thumbnails
product thumbnail on the invoice/receipt page using the og image and not the thumbnail version

### Notes
**Possible Cause**
I did some poking around the code base and I think the issue might be coming from these React Components ([media.js](https://github.com/reactioncommerce/reaction/blob/f40ff536c139d70da02ca10ae12655247452d658/imports/plugins/core/ui/client/components/media/media.js#L88) & [mediaGallery.js](https://github.com/reactioncommerce/reaction/blob/f40ff536c139d70da02ca10ae12655247452d658/imports/plugins/core/ui/client/containers/mediaGallery.js#L87)) I'm still pretty new to RC so I might be totally wrong about this.

**Possible related issue**
#3399 

**Images I used**
bike
shoe

### Versions
```shell
Node: 9.4.0
NPM: 5.6.0
Meteor Node: 8.9.3
Meteor NPM: 5.5.1
Reaction CLI: 0.26.4
Reaction: 1.7.0
Reaction branch: master
```

### PR DESCRIPTION
**Pull request by kieckhafer**
Resolves #3637
Impact: minor
Type: fix + refactor + performance

## Issue
Throughout the Reaction application, product images in various locations were being displayed at their full resolution, instead of their optimized versions (`large`, `medium`, `small`, `thumbnail`). It seems only the cart and Product grid are correctly using optimized versions, all other product images are using the raw uploaded image.

## Solution
This PR updates all product images throughout the application to use the optimized versions. In addition, this PR also updates the order fulfillment workflow, and the order confirmation screen, to use our `ProductImage` component, instead of just using a regular  tag.

## Breaking changes
- None

## Testing
1. Login as an administrator, and create one or more products (or update the basic reaction product) to have at least two different variants with images, and one without an image.
2. Open the inspector, and inspect the element of the product image everywhere you see it:

**On the Product Grid, with all three sizes of PDP options**

sweet_tee_s
sweet_tee_s
sweet_tee_s

**On the Grid Settings Dashboard**

sweet_tee_s

**On the PDP, both on the larger Media Gallery, and the smaller Options / Variant images**

sweet_tee_s

**Add all product variants to your cart, and inspect the cart images**

blank_image_and_blank_image

**Make an order with all product variants, and inspect the order Confirmation screen**

blank_image

**On the orders dashboard, open up the larger view and inspect the images on the large list view**

blank_image

**Open up the Orders fulfillment screen, and inspect the images in that flow both on the "normal" invoice, and also on the "checkable" invoice**
blank_image
blank_image

### CODE REVIEW

File: imports/plugins/core/orders/client/components/productImage.js [hunk 1]
```diff
@@ -31,7 +32,12 @@ class ProductImage extends Component {
     let mediaUrl;
 
     if (displayMedia(item)) {
-      mediaUrl = displayMedia(item).url();
+      const rawMediaUrl = displayMedia(item).url();
+      mediaUrl = rawMediaUrl;
+
+      if (size) {
+        mediaUrl = `${rawMediaUrl}?source=${size}`;
```
[the remark this answers is not in the document]
Comment (impactmass) [reply imports/plugins/core/orders/client/components/productImage.js]: thought to quickly ask that before I test in the UI

**Comment by kieckhafer** [comment]
@impactmass good catch, it's store. Ready for a look.

**Comment by kieckhafer** [comment]
@impactmass updated based on your notes.
1) Good catch, they should be the same size
2a) Updated to use `&` instead of `?`
2b) Updated to thumbnail, although they were set to `small` before I started the PR. It's probably good to update unless there was a reason they were small instead.

File: imports/plugins/included/default-theme/client/styles/cart/checkout.less [hunk 2]
```diff
@@ -230,7 +230,6 @@
 .order-details-media {
   width: 40px;
   height: 40px;
-  border: solid 1px #cccccc;
```
Comment (impactmass) [inline imports/plugins/included/default-theme/client/styles/cart/checkout.less]: Is removing the border here a new design direction we're taking? I'd thought we wanted the border around the images?

screenshot_2018-02-15_14_37_03_and_screenshot_2018-02-15_14_37_12_and_screenshots_and_test

**Comment by kieckhafer** [comment]
@impactmass I removed that border on purpose for 2 reasons.

1) This was the only place we were using the border, even the thumbnails in the orders panel don't use them
2) They didn't look good - the were cut off on one side, and very close to the image itself. We can mess with it to make them look better, but removing them was the easy option. @rymorgan thoughts?

basic_reaction_product

**Comment by rymorgan** [comment]
Yea, I think for consistencies sake you remove the borders.

**Pull request merged**